espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/InvocationOperation.java

```
@@ -0,0 +1,376 @@
+package io.appium.espressoserver.lib.helpers;
+
+import android.util.Log;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicReference;
+
+import androidx.test.platform.app.InstrumentationRegistry;
+import io.appium.espressoserver.lib.model.BackdoorResultVoid;
+
+//https://github.com/calabash/calabash-android-server/blob/develop/server/app/src/androidTest/java/sh/calaba/instrumentationbackend/query/InvocationOperation.java
+
+public class InvocationOperation {
+    private final String methodName;
+    private final List<?> arguments;
+    private final Class<?>[] argumentTypes;
+
+    public InvocationOperation(String methodName, List<?> arguments) {
+        this.methodName = methodName;
+        this.arguments = arguments;
+        this.argumentTypes = parseArgumentTypes(arguments);
+    }
+
+
+    public Object apply(final Object o) throws Exception {
+        Log.d("Backdoor Method", getName());
+        Log.d("Invoking on ", o.toString());
+        final AtomicReference<Object> ref = new AtomicReference<Object>();
+        final AtomicReference<Exception> refEx = new AtomicReference<Exception>();
+
+        InstrumentationRegistry.getInstrumentation().runOnMainSync(new Runnable() {
+            @Override
+            public void run() {
+                MethodWithArguments method = findCompatibleMethod(o);
+
+                if (method == null) {
```

I would rather throw an exception inside `findCompatibleMethod`

---

added